### PR TITLE
Update test matrix according to changes to PL-Qiskit plugin

### PR DESCRIPTION
**Pull request by austingmhuang**
The PL-Qiskit plugin no longer has tests that require an ibmq account. Namely `test_ibmq.py` and `test_runtime.py` have been deleted from the plugin CI tests and so are not relevant for the test matrix. This update accounts for the deletion of these tests and makes changes accordingly.

Related Issue:
[sc-69774]

### CODE REVIEW

File: .github/workflows/qiskit-latest-latest.yml
Comment (austingmhuang) [inline .github/workflows/qiskit-latest-latest.yml:1]: these 5 .github files are changed when `compile.py` is changed and ran with `python3 compile`.

File: compile.py [hunk 1]
```diff
@@ -15,7 +15,6 @@
             "--device=qiskit.aer --tb=short --skip-ops --shots=None --device-kwargs backend=statevector_simulator",
             "--device=qiskit.aer --tb=short --skip-ops --shots=None --device-kwargs backend=unitary_simulator",
         ],
-        "test_kwargs": ["-k 'not test_ibmq.py and not test_runtime.py'"],
```
Comment (austingmhuang) [inline compile.py:18]: these tests no longer exist in the plugin repo.

File: README.md [hunk 2]
```diff
@@ -33,16 +33,6 @@ All entries in the matrix are tested against PennyLane latest (GitHub master).
...
-| :----------------------------------------------------------------- | :--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| [Qiskit-IBMQ](https://github.com/PennyLaneAI/pennylane-qiskit)     | [![GitHub Workflow Status](https://img.shields.io/github/actions/workflow/status/PennyLaneAI/pennylane-qiskit/ibmq_tests.yml?branch=master)](https://github.com/PennyLaneAI/pennylane-qiskit/actions/workflows/ibmq_tests.yml)     |
-| [Qiskit-IBMQ 1.0](https://github.com/PennyLaneAI/pennylane-qiskit) | [![GitHub Workflow Status](https://img.shields.io/github/actions/workflow/status/PennyLaneAI/pennylane-qiskit/ibmq_tests_1.yml?branch=master)](https://github.com/PennyLaneAI/pennylane-qiskit/actions/workflows/ibmq_tests_1.yml) |
-
```
Comment (austingmhuang) [inline README.md:45]: Seems irrelevant to me now that ibmq tests have been removed. Alternatively we can replace this note by saying that the Qiskit tests now only run on local simulators.

**Review by soranjh — COMMENTED** [review]
Seems that all the tests are failing.

File: .github/workflows/qiskit-latest-latest.yml
Comment (austingmhuang) [reply .github/workflows/qiskit-latest-latest.yml:1]: These 5 workflow files are automatically generated by running `python3 compile` as indicated in the [readme](https://github.com/PennyLaneAI/plugin-test-matrix?tab=readme-ov-file#adding-a-plugin). We manually change `compile.py`, which when executed, changes those 5 workflow files.

**Comment by austingmhuang** [comment]
> Seems that all the tests are failing.

The tests failing for braket are not related to this PR. The qiskit tests seem to be failing due to some test regarding the noise models:
```
FAILED plugin_repo/tests/test_converter.py::TestLoadNoiseModel::test_build_noise_model - TypeError: ThermalRelaxationError.__init__() got an unexpected keyword argument 'tg'
```

Maybe @obliviateandsurrender has some insight as to what's happening?

**Comment by obliviateandsurrender** [comment]
Please check the pennylane version for which the test matrix is being tested. Thermal relaxation default argument was corrected to `tg` from `tq` in the master.

File: README.md [hunk 3]
```diff
@@ -33,16 +33,6 @@ All entries in the matrix are tested against PennyLane latest (GitHub master).
 * The Braket plugin device integration tests are run with `-k “not Sample and not no_0_shots”`,
   see #6
 
-* The Qiskit tests above are run with `-k 'not test_ibmq.py and not test_runtime.py'`, that is,
```
Comment (obliviateandsurrender) [inline README.md:36]: Maybe we should leave a note here that IBM Quantum backends are no longer tested? 🤔

Comment (austingmhuang) [reply README.md:36]: Well we never tested IBM Quantum backends in the first place, we were using the cloud simulator. I guess we could still keep a note about how this doesn't test the actual backends?

**Review by obliviateandsurrender — APPROVED** [review]

**Review by soranjh — APPROVED** [review]

**Review by Alex-Preciado — APPROVED** [review]
